packages/flame/lib/src/components/core/component.dart

```diff
@@ -462,6 +462,10 @@ class Component {
     _children?.forEach((c) => c.updateTree(dt));
   }
 
+  /// This method will be invoked from lifecycle if [component] has been added
+  /// to or removed from its parent children list.
+  void onChildrenChanged(Component component) {}
```

Comment:
I think it would be nice to add an enum here that says whether the component was added or removed, so that the user doesn't have to look at the parent to know that. I would also rename `component` to `child` here.

Reply:
Thanks for the feedback. How about that enum:
```
ComponentOnChildrenChangedType { add, remove }
```
and added in component.dart ?

Reply:
I think `ChildrenChangedType` is enough. Yeah we can have it directly in `component.dart` for now, just remember to export it in the barrel-file.